src/khoj/database/adapters/__init__.py

```diff
@@ -423,12 +444,18 @@ def get_default_agent():
 
     @staticmethod
     def create_default_agent():
-        # First delete the existing default
-        Agent.objects.filter(name=AgentAdapters.DEFAULT_AGENT_NAME).delete()
-
         default_conversation_config = ConversationAdapters.get_default_conversation_config()
         default_personality = prompts.personality.format(current_date="placeholder")
 
+        if Agent.objects.filter(name=AgentAdapters.DEFAULT_AGENT_NAME).exists():
+            agent = Agent.objects.filter(name=AgentAdapters.DEFAULT_AGENT_NAME).first()
```

Comment:
We could use the `get_agent_by_slug` method from above here?

```suggestion
        if get_agent_by_slug(AgentAdapters.DEFAULT_AGENT_SLUG).exists():
            agent = get_agent_by_slug(AgentAdapters.DEFAULT_AGENT_SLUG).first()
```